### PR TITLE
Keys api

### PR DESCRIPTION
**Pull request by emmanuelantony2000**
## PR Checklist

- [x] I have read and signed the [Contributor License Agreement](https://forms.gle/PZyiY5N2GDQU8GsV9).

### CODE REVIEW

**Comment by emmanuelantony2000** [comment]
I have to yet do the scoped search key, more [here](https://typesense.org/docs/0.19.0/api/api-keys.html#generate-scoped-search-key).

File: src/client/keys.rs [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+}
+
+impl<'a, T> ClientKeys<'a, T>
+where
+    T: HttpLowLevel,
+{
+    /// Create an API Key.
+    pub async fn create(
+        &self,
+        actions: Vec<String>,
```
Comment (morenol) [inline src/client/keys.rs:18]: I think that this should be an enum type, or T: AsRef<str>,  and create constants for the values defined in https://typesense.org/docs/0.19.0/api/api-keys.html#sample-actions

**Review by morenol — APPROVED** [review]
lgtm, maybe we could add some tests mocking the responses from the documentation

**Comment by emmanuelantony2000** [comment]
Yeah just wanted to confirm if the design and code is ok, I'll add the tests and documentation and finalize it.

File: src/client/keys.rs [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+        filter_by: impl AsRef<str>,
+        expires_at: usize,
+    ) -> crate::Result<String> {
+        let generate_scoped_search_key = GenerateScopedSearchKey {
+            filter_by: filter_by.as_ref().to_string(),
+            expires_at,
+        };
+        let params = serde_json::to_string(&generate_scoped_search_key)?;
+
+        let mut mac = Hmac::<Sha256>::new_from_slice(key.as_ref().as_bytes()).unwrap();
```
Comment (morenol) [inline src/client/keys.rs:91]: I think that we should create an issue to remove the `unwraps` from the code.

**Pull request merged**